server/src/main/java/org/elasticsearch/health/HealthStatus.java

```diff
@@ -29,6 +30,10 @@ public enum HealthStatus implements Writeable {
         this.value = value;
     }
 
+    public static HealthStatus read(StreamInput in) throws IOException {
```

Comment:
this mix of byte value and ordinal (the read/writeEnum methods use ordinal) is very brittle. 

I think we should read the byte value (since we're writing that) on the wire (the byte value currently corresponds with the ordinal value of every entry so it works 

Shall we also add an `AbstractSerializationTestCase` for this class?

Reply:
Good catch! I will fix it and the test.